gobblin-api/src/main/java/org/apache/gobblin/publisher/DataPublisher.java

```diff
@@ -125,4 +125,8 @@ public boolean canBeSkipped() {
   protected boolean shouldPublishMetadataFirst() {
     return true;
   }
+
+  public boolean submitLineageEvent (Collection<? extends WorkUnitState> states) {
```

Comment:
Please name boolean methods with yes/no questions.

Reply:
I guess you can throw an exception as a way to indicate failure instead of returning `false`